### PR TITLE
feat(python,rust): Add `dt.to_string` alias for `dt.strftime`

### LINKED ISSUE #8215: Rename `str.strptime`/`dt.strftime` methods
These names are well-established historically, but definitely do not follow our naming conventions.

I would suggest:
* `str.strptime` -> `str.to_timestamp`
* `dt.strftime` -> `dt.to_string`

There might be better names - suggestions?

### PR DESCRIPTION
**Pull request by stinodego**
Closes #8215 

Changes:
* Add `to_string` to the temporal namespace. `strftime` is now an alias for this method - we give users the option to use the familiar `strftime`, but internally we use the more readable `to_string`.
* Update references, docstrings, etc.

I'm planning to make the `format` parameter optional, and move the logic for determining a default format from `cast` to this method, and have `cast` call this method. But I'll do that refactor in another PR.

### CODE REVIEW

File: polars/polars-core/src/chunked_array/temporal/date.rs [hunk 1]
```diff
@@ -30,8 +30,9 @@ impl DateChunked {
         Int32Chunked::from_vec(name, unit).into()
     }
 
-    /// Format Date with a `format` rule. See [chrono strftime/strptime](https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html).
-    pub fn strftime(&self, format: &str) -> Utf8Chunked {
+    /// Convert from Date to Utf8 with the given format.
+    /// See [chrono strftime/strptime](https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html).
+    pub fn to_string(&self, format: &str) -> Utf8Chunked {
```
Comment (ritchie46) [inline polars/polars-core/src/chunked_array/temporal/date.rs:35]: Why are these renamed? 

Shouldn't we just be able to add 2 aliases?

File: polars/polars-core/src/series/implementations/dates_time.rs [hunk 2]
```diff
@@ -308,7 +308,7 @@ macro_rules! impl_dyn_series {
                         .into_series()
                         .date()
                         .unwrap()
-                        .strftime("%Y-%m-%d")
+                        .to_string("%Y-%m-%d")
```
Comment (ritchie46) [inline polars/polars-core/src/series/implementations/dates_time.rs:311]: Which then makes these changes unneeded.

File: polars/polars-core/src/chunked_array/temporal/date.rs (hunk 1 above)
Comment (stinodego) [reply polars/polars-core/src/chunked_array/temporal/date.rs:35]: We can! I had it like that initially, but aren't these always accessed through the namespace? In that case, we only need the alias on the namespace level (which is how it is now).

But if these methods on the chunked array are called directly by users then we should keep the `strftime` method.

File: polars/polars-core/src/series/implementations/dates_time.rs (hunk 2 above)
Comment (stinodego) [reply polars/polars-core/src/series/implementations/dates_time.rs:311]: I thought it would be nice to use the more readable / more ideally named version internally, and keep `strftime` for Polars users who have familiarity with that name.

But indeed, technically we can keep using `strftime` here if we keep those methods on the chunked arrays.

File: polars/polars-core/src/chunked_array/temporal/date.rs (hunk 1 above)
Comment (stinodego) [reply polars/polars-core/src/chunked_array/temporal/date.rs:35]: @ritchie46 Could you get back to me on this one? I will update the PR accordingly.

File: polars/polars-core/src/series/implementations/dates_time.rs (hunk 2 above)
Comment (ritchie46) [reply polars/polars-core/src/series/implementations/dates_time.rs:310]: RUST SIDE:

Do we now have `strftime` and `to_string` as expression? I think we should keep both on the expression side to keep consistency with python. 

In the lower level implementation, I think it's fine to only keep `to_string`.

**Comment by ritchie46** [comment]
I believe we still miss a docs entry (maybe I missed it).

File: polars/polars-core/src/series/implementations/dates_time.rs (hunk 2 above)
Comment (stinodego) [reply polars/polars-core/src/series/implementations/dates_time.rs:310]: > RUST SIDE:
> 
> Do we now have `strftime` and `to_string` as expression? I think we should keep both on the expression side to keep consistency with python.

We do! `DateLikeNameSpace` has both `to_string` and `strftime` implemented.

And for Series, `TemporalMethods` also has both. So that should cover it.

**Comment by stinodego** [comment]
> I believe we still miss a docs entry (maybe I missed it).

Oops, I missed that one. Added now.

_(I'd swear the CI used to fail when there were undocumented methods...)_

**Comment by stinodego** [comment]
@ritchie46 I think this can be merged - or do you have any doubts?

**Comment by stinodego** [comment]
@ritchie46 gentle ping :) As I said, if you have any doubts, I don't mind addressing them.

**Comment by ritchie46** [comment]
Seems good. Architecture seems the same. As long as `strftime` is still available on all levels, feel free to merge. :)

**Comment by stinodego** [comment]
All right, then I'm pressing the green button :)
